core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/OracleExportAction.java

```diff
@@ -0,0 +1,235 @@
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+/**
+ * Action that exports data from Oracle.
+ */
+@Plugin(type = Action.PLUGIN_TYPE)
+@Name("OracleExport")
+@Description("Action to export data from Oracle")
+public class OracleExportAction extends Action {
+  private final OracleExportActionConfig config;
+  private StringBuilder scriptContent = null;
+  private String execute = null;
+  private static final Set<String> FORMATS = ImmutableSet.of("csv", "tsv", "psv");
+
+  public OracleExportAction(OracleExportActionConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void run(ActionContext context) throws Exception {
+    init();
+    Connection connection = new Connection(config.oracleServerHostname, config.oracleServerPort);
+    try {
+      connection.connect();
+      boolean isAuthenticated = connection.authenticateWithPassword(config.oracleServerUsername,
+                                                                    config.oracleServerPassword);
+      if (isAuthenticated == false) {
+        throw new IOException(String.format("SSH authentication error when connecting to %s@%s on port %d",
+                                            config.oracleServerUsername, config.oracleServerHostname,
+                                            config.oracleServerPort));
+      }
+      Session session = connection.openSession();
+      session.execCommand(execute);
+      InputStream stdout = new StreamGobbler(session.getStdout());
+      BufferedReader outBuffer = new BufferedReader(new InputStreamReader(stdout, Charsets.UTF_8));
+      Integer exitCode = session.getExitStatus();
+      if (exitCode != null && exitCode != 0) {
+        throw new IOException(String.format("Error:command %s running on hostname %s finished with exit code: %d ",
+                                            execute, config.oracleServerHostname, exitCode));
+      }
+      String out = CharStreams.toString(outBuffer);
+      //SQLPLUS command errors are not fetched from session.getStderr().
+      //Errors and output received after executing the command in SQLPlus prompt are the one that
+      //are printed on the SQL prompt
+      if (out.contains("ERROR at line")) {
+        throw new IOException(String.format("Error:executing sqlplus query %s on hostname %s with error message: %s",
+                                            config.queryToExecute, config.oracleServerHostname, out));
+      }
+      Path file = new Path(config.pathToWriteFinalOutput);
+      FileSystem fs = FileSystem.get(file.toUri(), new Configuration());
+      FSDataOutputStream outStream = fs.create(file);
```

Comment:
FSDataOutputStream and BufferedWriter should be closed using try with resources.